src/otx/core/data/module.py

```diff
@@ -62,18 +62,15 @@ def __init__(
         auto_num_workers: bool = False,
         device: DeviceType = DeviceType.auto,
         input_size: int | tuple[int, int] | None = None,
+        adaptive_input_size: Literal["auto", "downscale", "none"] = "none",
+        input_size_multiplier: int = 1,
     ) -> None:
         """Constructor."""
         super().__init__()
         self.task = task
         self.data_format = data_format
         self.data_root = data_root
 
-        if input_size is not None:
-            for subset_cfg in [train_subset, val_subset, test_subset, unlabeled_subset]:
-                if subset_cfg.input_size is None:
-                    subset_cfg.input_size = input_size
```

Comment:
I don't understand this, recipe updates, and `how torchvision transform factor applies $(input_size) is slightly changed` in your summary.
In original impl, if users want to set different `input_size` only for validation, the users can just set an additional `input_size` in the validation section.
But your update doesn't consider `input_size` as the global parameter.
I think that subset config manually set by the users in the recipe must have a higher priority.

Reply:
I want to keep this policy in this PR but there is no way to distinguish between input_size user sets and existing input_size. For example, when using yolox_tiny model which sets input_size to val_subset and test_subset, user need to remove those values in the recipe to use input_size because user's input_size can't be applied with previous policy. So, I have thought how to deal with it, there is no way except current change. If you have some idea of this, please tell me. I can try it instead of this :)

Reply:
I think it doesn't matter that users remove those values for their custom configurations.
It is their responsibility because we just support our optimal configurations.

Reply:
Of course, I'm not sure if it is possible, but we can update `input_size` in valset and testset to `$(input_size) * 0.65` in yolox_tiny.

Reply:
thanks for sharing your idea. I updated it as you said. https://github.com/openvinotoolkit/training_extensions/pull/3788/commits/8d34d2c3c0cf5891865238881f92686df1075b8e